src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -1130,6 +1131,14 @@ set_cu_read_range(cuidx_type cuidx, uint32_t start, uint32_t size)
     throw xrt_core::error(ret, "failed to set cu read range");
 }
 
+std::shared_ptr<xrt_core::graph_handle>
+device_linux::
+open_graph_handle(const xrt::uuid& xclbin_id, const char* name, xrt::graph::access_mode am)
+{
+   return std::make_shared<ZYNQ::shim::graph_object>(
+                  static_cast<ZYNQ::shim*>(get_device_handle()), xclbin_id, name, am);
+}
```

Comment:
Same ownership question.  This one should definitely return a std::unique_ptr since it creates the object here.